internal/ocean/ofas/deploy.go

```diff
@@ -35,20 +37,74 @@ func ValidateClusterContext(ctx context.Context, client kubernetes.Interface, cl
-	if err != nil && !k8serrors.IsAlreadyExists(err) {
-		return fmt.Errorf("could not create deployer service account, %w", err)
-	}
+	for {
+		// Decode the YAML into an unstructured object
+		var obj unstructured.Unstructured
+		if err := decoder.Decode(&obj); err == io.EOF {
+			break
+		} else if err != nil {
+			return fmt.Errorf("could not decode yaml, %w", err)
+		}
+
+		// Handle each resource type
+		switch obj.GetKind() {
+		case "Namespace":
+			ns := &corev1.Namespace{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, ns); err != nil {
+				return fmt.Errorf("could not decode namespace yaml, %w", err)
+			}
+			_, err := client.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+			if err != nil && !k8serrors.IsAlreadyExists(err) {
+				return fmt.Errorf("could not create namespace, %w", err)
+			}
+
+		case "ServiceAccount":
+			sa := &corev1.ServiceAccount{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, sa); err != nil {
+				return fmt.Errorf("could not decode service account yaml, %w", err)
+			}
+			_, err := client.CoreV1().ServiceAccounts(sa.Namespace).Create(ctx, sa, metav1.CreateOptions{})
+			if err != nil && !k8serrors.IsAlreadyExists(err) {
+				return fmt.Errorf("could not create service account, %w", err)
+			}
+
+		case "RoleBinding":
+			rb := &rbakv1.RoleBinding{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, rb); err != nil {
+				return fmt.Errorf("could not decode role binding yaml, %w", err)
+			}
+			_, err := client.RbacV1().RoleBindings(rb.Namespace).Create(ctx, rb, metav1.CreateOptions{})
+			if err != nil && !k8serrors.IsAlreadyExists(err) {
+				return fmt.Errorf("could not create role binding, %w", err)
+			}
+
+		case "ClusterRole":
+			cr := &rbakv1.ClusterRole{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, cr); err != nil {
+				return fmt.Errorf("could not decode cluster role yaml, %w", err)
+			}
+			_, err := client.RbacV1().ClusterRoles().Create(ctx, cr, metav1.CreateOptions{})
+			if err != nil && !k8serrors.IsAlreadyExists(err) {
+				return fmt.Errorf("could not create cluster role, %w", err)
+			}
 
-	_, err = client.RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
-	if err != nil && !k8serrors.IsAlreadyExists(err) {
-		return fmt.Errorf("could not create deployer cluster role binding, %w", err)
+		case "ClusterRoleBinding":
+			crb := &rbakv1.ClusterRoleBinding{}
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, crb); err != nil {
+				return fmt.Errorf("could not decode cluster role binding yaml, %w", err)
+			}
+			_, err := client.RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
+			if err != nil && !k8serrors.IsAlreadyExists(err) {
+				return fmt.Errorf("could not create cluster role binding, %w", err)
+			}
+		}
 	}
```

Comment:
This looks fine to me.

I've not written anything similar myself. This very much looks like a `kubectl apply` operation that is good.

Perhaps we should consider having a `default` step there where we perhaps either log an error where the template has a resource that is not handled specifically or have it fail completely. Might help us detect a mismatch between the template and the deploy method.